File: src/test/java/ukalus/math/VectorTest.java
/*
 * Created on Feb 21, 2005
 *
 */
package ukalus.math;

import java.util.HashMap;
import java.util.Map;

import junit.framework.TestCase;

/**
 * @author pmgremo
 * 
 */
public class VectorTest extends TestCase {

  public void testRotate() {
    Vector actual = new Vector(1, 0).rotate(90);
    assertEquals(0, (int) actual.getX());
    assertEquals(-1, (int) actual.getY());
  }

  public void testFindInHashMap() {
    Map<Vector, Object> map = new HashMap<Vector, Object>();
    map.put(new Vector(1, 0), null);
    assertTrue(map.containsKey(new Vector(1, 0)));
  }

  public void testAdd() {
    Vector result = new Vector(1, 0);
    Vector angle = new Vector(1, 1);
    result = result.add(angle);
    assertEquals(new Vector(2, 1), result);
  }

  public void testSubtract() {
    Vector result = new Vector(1, 0);
    Vector angle = new Vector(1, 1);
    result = result.subtract(angle);
    assertEquals(new Vector(0, -1), result);
  }

}

File: src/test/java/ukalus/math/VectorTest.kt
/*
 * Created on Feb 21, 2005
 *
 */
package ukalus.math

import java.util.HashMap

import junit.framework.TestCase

/**
 * @author pmgremo
 */
class VectorTest : TestCase() {

    fun testRotate() {
        val actual = Vector(1.0, 0.0).rotate(90.0)
        TestCase.assertEquals(0, actual.x.toInt())
        TestCase.assertEquals(-1, actual.y.toInt())
    }

    fun testFindInHashMap() {
        val map = HashMap<Vector, Any?>()
        map.put(Vector(1.0, 0.0), null)
        TestCase.assertTrue(map.containsKey(Vector(1.0, 0.0)))
    }

    fun testAdd() {
        var result = Vector(1.0, 0.0)
        val angle = Vector(1.0, 1.0)
        result = result.add(angle)
        TestCase.assertEquals(Vector(2.0, 1.0), result)
    }

    fun testSubtract() {
        var result = Vector(1.0, 0.0)
        val angle = Vector(1.0, 1.0)
        result = result.subtract(angle)
        TestCase.assertEquals(Vector(0.0, -1.0), result)
    }

}